src/modules/menu/components/menu/reportProblem/ReportProblemButton.vue

```diff
@@ -0,0 +1,215 @@
+<script setup>
+import { computed, nextTick, ref } from 'vue'
+
+import sendFeedback from '@/api/feedback.api'
+import { createShortLink } from '@/api/shortlink.api'
+import HeaderLink from '@/modules/menu/components/header/HeaderLink.vue'
+import FeedbackActionButtons from '@/modules/menu/components/menu/common/FeedbackActionButtons.vue'
+import EmailValidationField from '@/utils/components/EmailValidationField.vue'
+import ImportLocalFile from '@/utils/components/ImportLocalFile.vue'
+import ModalWithBackdrop from '@/utils/components/ModalWithBackdrop.vue'
+import log from '@/utils/logging'
+import { isValidEmail } from '@/utils/utils'
+
+const acceptedFileTypes = '.kml,.gpx,.pdf,.zip,.jpg,.jpeg,.kmz'
+
+const props = defineProps({
+    showAsLink: {
+        type: Boolean,
+        default: false,
+    },
+})
+
+// Reactive data
+const feedbackMessageTextArea = ref(null)
+const requestResults = ref(null)
+const reportProblemCloseSuccessful = ref(null)
+
+const showReportProblemForm = ref(false)
+const userIsTypingFeedback = ref(false)
+const feedback = ref({
+    message: null,
+    kml: null,
+    email: null,
+    file: null,
+})
+const request = ref({
+    pending: false,
+    failed: false,
+    completed: false,
+})
+const shortLink = ref('')
+
+//  Computed properties
+const feedbackCanBeSent = computed(() => {
+    return !request.value.pending && isEmailValid.value && isMessageValid.value
+})
+const isMessageValid = computed(() => feedback.value.message?.length > 0)
+const isEmailValid = computed(() => {
+    return !feedback.value.email || isValidEmail(feedback.value.email)
+})
+
+// Methods
+async function sendReportProblem() {
+    // if the request was already sent, we don't allow the user to double send
+    if (request.value.completed) {
+        // we instead close the modal if he/she clicks on the check mark
+        closeAndCleanForm()
+        return
+    }
+
+    request.value.pending = true
+    try {
+        const feedbackSentSuccessfully = await sendFeedback(
+            '[web-mapviewer] Problem report', // subject
+            feedback.value.message,
+            {
+                email: feedback.value.email,
+                attachment: feedback.value.file,
+            }
+        )
+        request.value.completed = feedbackSentSuccessfully
+        request.value.failed = !feedbackSentSuccessfully
+    } catch (err) {
+        log.error('Error while sending feedback', err)
+        request.value.failed = true
+    } finally {
+        request.value.pending = false
+    }
+    await nextTick()
+    // scrolling down to make sure the message with request results is visible to the user
+    if (request.value.failed) {
+        requestResults.value.scrollIntoView()
+    } else if (request.value.completed) {
+        reportProblemCloseSuccessful.value.focus()
```

Comment:
The focus on success doesn't work on the test link using chrome

Reply:
Hmm, very strange, it works for me as seen below. I use keyboard to fill up the form, tab to move, and enter/space to press the button.
![Peek 2024-04-04 15-53 - auto focus on close button](https://github.com/geoadmin/web-mapviewer/assets/1421861/3508e8e6-474e-4b55-a815-f53072003a45)

I tried on Chrome and Firefox, and I have a test case for that also: https://github.com/geoadmin/web-mapviewer/pull/624/files#diff-60740a7e3043624f1fc8b73fa41349cc88bdf6ad1131cf3f3575bcdf73cd7a25R100